BL-1714: Allow override of pcavailability param.

dkinzer opened the pull request:
If user submits pcavailability value via a param, use that instead of the
default set in the Primo.configuration.

ebtoner commented:
@dkinzer -- pcAvailability is a boolean paramter, so the only expected values are true and false. So the default in Library Search is "false" so the only value that should be allowed for the overrride is "true". 

See https://developers.exlibrisgroup.com/primo/apis/docs/primoSearch/R0VUIC9wcmltby92MS9zZWFyY2g=/

dkinzer commented:
@ebtoner that's a very good point but the reason I use "foo" instead of a boolean in the test case is so that am absolutely sure that I am able to override that parameter.

ebtoner commented:
I'm not sure that I understand your logic. Shouldn't the override fail with "foo" since it is not an accepted value for the Primo Search?

dkinzer commented:
@ebtoner this library doesn't provide the feature of erroring out if the user of the library provides a bogus value for one of the available parameters. So I can't test for that unless I'm adding that feature to the library.

The feature I'm trying to test here is that the user can override a default parameter using a url parameter override.  Before this change this library only used the default configured via Primo.configuration.

I've also added a new feature because I noticed on further using this library in tul_cob that if the url key parameter presents as a key instead of symbol then the override does not take because the symbold one (which is the default key type) gets precedent. 

I added a new test for this new feature to make sure that if the parameter key is of type string or of type symbol it will still use the override.
